src/parser.rs

```diff
@@ -516,7 +516,18 @@ impl<'a> Parser<'a> {
                         }
                     };
                 self.expect_token(&Token::RParen)?;
-                Ok(expr)
+                if !self.consume_token(&Token::Period) {
```

Comment:
I am thinking if the `CompositeAccess` can only appear after a `)` -- it seems like it should be possible in other places, though i suppose without a parenthesis `.<name>` will appear as a compound identifier.